src/game/ticTacToeGame.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+exports.__esModule = true;
+exports.TicTacToe = void 0;
+var events_1 = require("../events/events");
+var types_1 = require("../events/types");
+var gameTypes_1 = require("../shared/types/gameTypes");
+var playerTypes_1 = require("../shared/types/playerTypes");
+var TicTacToe = /** @class */ (function () {
+    function TicTacToe(toMove, gameType) {
+        if (toMove === void 0) { toMove = playerTypes_1.PlayerTypes.player1; }
+        if (gameType === void 0) { gameType = gameTypes_1.GameTypes.twoPlayer; }
+        this.gameType = gameTypes_1.GameTypes[gameType];
+        this.piecesEnum = {
+            x: "X",
+            o: "O"
+        };
+        this.eventsEnum = {
+            gameStart: 0,
+            player1Moved: 1,
+            player2Moved: 2,
+            aiMoved: 3,
+            gameOver: 4
+        };
+        this.toMove = playerTypes_1.PlayerTypes[toMove];
+        this.board = [
+            ["", "", ""],
+            ["", "", ""],
+            ["", "", ""],
+        ];
```

Comment:
We could create a `createEmptyBoard` function (in a separate file) that gets this hard-coded stuff out of this file.